### PR TITLE
fix(ci): Use unique keys for React and Storybook build cache

### PR DESCRIPTION
**Pull request by DafyddLlyr**
image

Fixes issue where only Storybook build was getting passed to Vultr, as a result of [the recent parallelization of CI jobs ](https://github.com/theopensystemslab/planx-new/pull/1154)

I've tried a few (now squashed) commits on this branch to make sure the cache is behaving as expected 👍

### CODE REVIEW

File: .github/workflows/pull-request-main.yml [hunk 1]
```diff
@@ -392,13 +392,13 @@ jobs:
...
           path: ./${{ env.EDITOR_DIRECTORY }}/build
-          key: ${{ runner.os }}-${{ hashFiles('editor.planx.uk/**', '!editor.planx/build/**') }}
+          key: ${{ runner.os }}-${{ hashFiles('editor.planx.uk/**', '!editor.planx/build/**') }}-react
       - name: Download Storybook build assets
         id: cache-storybook-assets
         uses: actions/cache@v3
         with:
           path: ./${{ env.EDITOR_DIRECTORY }}/build
-          key: ${{ runner.os }}-${{ hashFiles('editor.planx.uk/**', '!editor.planx/build/**') }}
+          key: ${{ runner.os }}-${{ hashFiles('editor.planx.uk/**', '!editor.planx/build/**') }}-storybook
```
Comment (jessicamcinchak) [inline .github/workflows/pull-request-main.yml:401]: should this match the change on L211? so `!editor.planx/build/storybook/**`??

**Review by jessicamcinchak — APPROVED** [review]
Thanks for this fix! One question, but CI links all working as expected :+1:

File: .github/workflows/pull-request-main.yml (hunk 1 above)
Comment (DafyddLlyr) [reply .github/workflows/pull-request-main.yml:401]: Good catch - L211 is me not cleaning up after myself properly from a previous commit sorry! 

Storybook is already build in the `/build/storybook` directory (in the Editor's `package.json` script `build-storybook`). We actually want Storybook to rebuild if there's any change in the Editor directory - the cache key should change if there's a been an update to any file in the `/editor.planx` directory.


Change pushed up!

**Comment by gunar** [comment]
Nice catch! Sorry! Thanks!

**Pull request merged**